src/magic/model/MagicAbility.java

```diff
@@ -1497,6 +1497,26 @@ protected void addAbilityImpl(final MagicAbilityStore card, final Matcher arg) {
             ));
         }
     },
+    WhenACounterIsPutOnPermanent("Whenever a (?<type>[^ ]+) counter is put on " + ARG.WORDRUN + ", " + ARG.EFFECT, 10) {
```

Comment:
please use an existing ARG constant instead of hardcoding one in the regex, ARG.WORD1 can be used here.

Reply:
Okay. I just copied from put counter action in `MagicRuleEventAction.java`

Reply:
I see, I will fix the put counter action to use ARG constant as well. Thanks for the info.